src/client/automation/playback/scroll.js

```diff
@@ -176,18 +204,42 @@ export default class ScrollAutomation {
 
         if (window.top !== window) {
             return sendRequestToFrame({
-                cmd:     SCROLL_REQUEST_CMD,
-                offsetX: currentOffsetX,
-                offsetY: currentOffsetY
+                cmd:             SCROLL_REQUEST_CMD,
+                offsetX:         currentOffsetX,
+                offsetY:         currentOffsetY,
+                maxScrollMargin: this.maxScrollMargin
             }, SCROLL_RESPONSE_CMD, window.parent);
         }
 
         return Promise.resolve();
     }
 
+    _isElementHiddenByFixed () {
+        var clientDimensions = positionUtils.getClientDimensions(this.element);
+        var elementInPoint   = positionUtils.getElementFromPoint(clientDimensions.left +
+                                                                 this.offsetX, clientDimensions.top + this.offsetY);
+
+        return elementInPoint && styleUtils.getComputedStyle(elementInPoint).position === 'fixed';
+    }
+
+    _isScrollMarginTooBig () {
+        var minWindowDimension = Math.min(styleUtils.getInnerWidth(window), styleUtils.getInnerHeight(window));
+
+        return this.maxScrollMargin >= minWindowDimension / 2;
+    }
+
     run () {
         this._scrollElement();
 
-        return this._scrollParents();
+        return this
+            ._scrollParents()
+            .then(() => whilst(() => !this._isScrollMarginTooBig() && this._isElementHiddenByFixed(), () => {
+                if (this._isScrollMarginTooBig() || !this._isElementHiddenByFixed())
```

Comment:
as I understand you check it in the `whilst` condition, why do you need to check it again?